plugins/out_s3/s3.c

```diff
@@ -1034,7 +1057,8 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
         create_time = time(NULL);
     }
 
-    ret = s3_put_object(ctx, tag, create_time, body, body_size);
+    ret = s3_put_object(ctx, tag, create_time, payload_buf, payload_size);
+    flb_free(payload_buf);
     if (ret < 0) {
```

Comment:
May be the issue is that in the non-compressed case- the `payload_size` and `payload_buf` variables are not set at all?? That's why I was thinking that we wouldn't change any of the code under the labels.